moped-etl/arcgis/components_to_agol.py

```diff
@@ -54,74 +63,192 @@ def make_esri_feature(*, esri_geometry_key, geometry, attributes):
+            # for point features, it is the buffered ring around the point as defined
+            # in the Moped component view
+            geometry = component.pop("geometry")
+            line_geometry = component.pop("line_geometry")
+
+            if not geometry:
+                continue
+
+            esri_geometry_key = get_esri_geometry_key(geometry)
+            feature = make_esri_feature(
+                esri_geometry_key=esri_geometry_key,
+                geometry=geometry,
+                attributes=component,
+            )
 
-        esri_geometry_key = get_esri_geometry_key(geometry)
-        feature = make_esri_feature(
-            esri_geometry_key=esri_geometry_key, geometry=geometry, attributes=component
-        )
+            # adds a special `source_geometry_type` column that will be useful on the `combined` layer
+            # so that we can keep track of if the original feature was a point or line geometry
+            feature["attributes"]["source_geometry_type"] = (
+                "point" if esri_geometry_key == "points" else "line"
+            )
 
-        # adds a special `source_geometry_type` column that will be useful on the `combined` layer
-        # so that we can keep track of if the original feature was a point or line geometry
-        feature["attributes"]["source_geometry_type"] = (
-            "point" if esri_geometry_key == "points" else "line"
-        )
+            if not feature:
+                continue
+
+            if esri_geometry_key == "points":
+                all_features["points"].append(feature)
+                # create the point -> line feature
+                if line_geometry["type"] == "LineString":
+                    # if we're converting a single point to a line, we need to convert that line geom to
+                    # a multi-line geometry
+                    line_geometry["coordinates"] = [line_geometry["coordinates"]]
+                line_feature = make_esri_feature(
+                    esri_geometry_key="paths",
+                    geometry=line_geometry,
+                    attributes=component,
+                )
+                all_features["combined"].append(line_feature)
+            else:
+                all_features["lines"].append(feature)
+                all_features["combined"].append(feature)
+
+        for feature_type in ["points", "lines", "combined"]:
+            logger.info(f"Processing {feature_type} features...")
+            features = all_features[feature_type]
+
+            logger.info("Deleting all existing features...")
+            delete_all_features(feature_type)
+
+            logger.info(
+                f"Uploading {len(features)} features in chunks of {UPLOAD_CHUNK_SIZE}..."
+            )
+            for feature_chunk in chunks(features, UPLOAD_CHUNK_SIZE):
+                logger.info("Uploading chunk....")
+                add_features(feature_type, feature_chunk)
+    else:
```

Comment:
This block looks only deletes and replaces features from projects that were updated since the last DAG run. That date will be passed when run by Airflow.